### DISCUSSION TITLE
Auth Identities: Email provider - unable to attain identity_data.email_verified = true

### DISCUSSION THREAD

**Discussion by jwhenshaw** [supabase/supabase#20191] [Questions] [answered]
Question: Should the `auth.identity.identity_data.email_verified` attribute be updated to `true` once the user successfully completes the change email flow?

Context 
----
I am wanting to update and verify a user's email and as in the future I want to associate multiple emails/identities with a single user I wish to use `auth.identities` to know what identities are related to a user.
Note, the project has Secure email change disabled.

Behaviour experienced (seems like a happy path):
----
1. From my React Native app using supabase-js I am able to trigger the change email flow using the `supabase.auth.updateUser` api which results in the confirm email change email being sent to the user,
2. a new `auth.identity` row is created in the db with the new email address and this record has `identity_data.email_verified` set to `false` (as expected),
3. the user then goes on to click the link within the email that takes them to a webapp, with the confirmation url as a url search param, that has a button - this is all to avoid [email prefetching issues](https://supabase.com/docs/guides/auth/auth-email-templates#email-prefetching) 
4. The user then presses the button and it successfully directs them to supabase's `/verify` endpoint, using the confirmation_url supabase provides in the email
5. Once the exchange with Supabase is done they are redirected back to another url successfully

At no point at there any auth logs within supabase with errors.

So, am wonder if am misunderstanding the intended use of the `identity_data.email_verified` attribute? Asking the question here as I can't find any specific documentation or previous discussion/issues that informs me and I've reached out to support.

Thanks in advance for any help

**Comment by jwhenshaw** [comment, marked as answer]
Supabase support got back to me with the following

> That email_verified field is set depending on the OAuth provider used. Unfortunately many OAuth providers don't actually give out this information, so the Auth team has tried to vet all of them whether it's in fact possible to create an account with an unverified email address.
There's no way to override this information right now. If the value is false, and you have email confirmations enabled for the project, the user will be sent a confirmation email _before_ they're allowed to sign in, but the identity and user account would be created.

Sharing here in case it helps anyone else and am going to close this but if anyone does have any workarounds or any more information please feel free to share it.

Thanks

**Reply by balfouri1210** [reply]
Thanks for sharing 👏
i was wonder why email_verified and phone_verified fields in the 'user_metadata' are not changed to 'true' after verification
